robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py

```diff
@@ -0,0 +1,532 @@
+#! /usr/bin/env python
+
+# System
+import math
+import sys
+import numpy as np
+
+# ROS
+import rospy
+import smach
+
+# TU/e Robotics
+import robot_smach_states as states
+import robot_smach_states.util.designators as ds
+from robot_skills.util import kdl_conversions
+from robot_skills.util.entity import Entity
+
+__all__ = ['FindPeople', 'FindFirstPerson', 'SetPoseFirstFoundPersonToEntity', 'FindPeopleInRoom']
+
+
+class FindPeople(smach.State):
+    """
+    Smach state to find a person. The robot looks around and tries to find
+    people in its view.
+
+    >>> # The requried imports on robot console
+    >>> import robot_smach_states as states
+    >>> import robot_smach_states.util.designators as ds
+    >>> from robot_skills.util.entity import Entity
+    >>>
+    >>> # Designator to store the result
+    >>> des = ds.VariableDesignator(resolve_type=[Entity])
+    >>>
+    >>> sm = state.FindPeople(robot=robot, properties={'id': 'NAME'}, found_people_designator=des.writeable)
+    >>> sm.execute()
+    >>> des.resolve()
+    Entity(id=something, )
+
+    """
+
+    def __init__(self, robot, properties=None, query_entity_designator=None,
+                 found_people_designator=None, look_distance=10.0, speak=False,
+                 strict=True, nearest=False, attempts=1, search_timeout=60,
+                 look_range=(-np.pi/2, np.pi/2), look_steps=8):
+        """ Initialization method
+        :param robot: robot api object
+        :param properties: (dict) (default: None) keyvalue pair of the properties a person must
+            possess. None as a value for a property would search for all possible
+            values of the property.
+        :param query_entity_designator: (default: None) An entity designator to match all found
```

Comment:
the query_entity_designator could use a better describtion. I only figured out what this variable does by looking at the code itself. Specify what type of entity it should be.